File: applicatie/src/main/java/com/vdzon/administratie/rest/gebruiker/GebruikerService.java
package com.vdzon.administratie.rest.gebruiker;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.vdzon.administratie.crud.UserCrud;
import com.vdzon.administratie.dto.GebruikerDto;
import com.vdzon.administratie.model.Gebruiker;
import com.vdzon.administratie.util.SessionHelper;
import com.vdzon.administratie.util.SingleAnswer;
import spark.Request;
import spark.Response;

import javax.inject.Inject;

public class GebruikerService {

    @Inject
    UserCrud crudService;

    protected Object postGebruiker(Request req, Response res) throws Exception {
        Gebruiker gebruiker = SessionHelper.getGebruikerOrThowForbiddenExceptin(req, crudService);
        String nieuweGebruikerJson = req.body();
        ObjectMapper mapper = new ObjectMapper();
        GebruikerDto nieuweGebruikerDto = mapper.readValue(nieuweGebruikerJson, GebruikerDto.class);
        Gebruiker originalGebruiker = crudService.getGebruiker(nieuweGebruikerDto.getUuid());
        if (originalGebruiker == null) {
            originalGebruiker = new Gebruiker();
        }
        Gebruiker updatedGebruiker = nieuweGebruikerDto.cloneGebruikerWithDtoFields(originalGebruiker);
        crudService.updateGebruiker(updatedGebruiker);
        return new SingleAnswer("ok");
    }

    protected Object removeGebruiker(Request req, Response res) throws Exception {
        Gebruiker gebruiker = SessionHelper.getGebruikerOrThowForbiddenExceptin(req, crudService);
        String gebruikerUuid = req.params(":uuid");
        if ("undefined".equals(gebruikerUuid)) {
            gebruikerUuid = null;
        }
        crudService.deleteGebruiker(gebruikerUuid);
        ;
        return new SingleAnswer("ok");
    }

    protected Object updatePassword(Request req, Response res) throws Exception {
        Gebruiker gebruiker = SessionHelper.getGebruikerOrThowForbiddenExceptin(req, crudService);
        String gebruikerUuid = req.params(":uuid");
        String newPassword = req.params(":newPassword");
        Gebruiker gebruikerToChange = crudService.getGebruiker(gebruikerUuid);
        Gebruiker changedGebruiker = gebruikerToChange.toBuilder().password(newPassword).build();
        crudService.updateGebruiker(changedGebruiker);
        return new SingleAnswer("ok");
    }

}

File: applicatie/src/main/java/com/vdzon/administratie/rest/gebruiker/GebruikerService.kt
package com.vdzon.administratie.rest.gebruiker

import com.fasterxml.jackson.databind.ObjectMapper
import com.vdzon.administratie.crud.UserCrud
import com.vdzon.administratie.dto.GebruikerDto
import com.vdzon.administratie.model.Administratie
import com.vdzon.administratie.model.Gebruiker
import com.vdzon.administratie.util.SessionHelper
import com.vdzon.administratie.util.SingleAnswer
import spark.Request
import spark.Response

import javax.inject.Inject
import java.util.ArrayList

class GebruikerService {

    @Inject
    lateinit internal var crudService: UserCrud

    @Throws(Exception::class)
    fun postGebruiker(req: Request, res: Response): Any {
        val gebruiker = SessionHelper.getGebruikerOrThowForbiddenExceptin(req, crudService)
        val nieuweGebruikerJson = req.body()
        val mapper = ObjectMapper()
        val nieuweGebruikerDto = mapper.readValue(nieuweGebruikerJson, GebruikerDto::class.java)
        var originalGebruiker: Gebruiker? = crudService!!.getGebruiker(nieuweGebruikerDto.uuid)
        if (originalGebruiker == null) {
            originalGebruiker = Gebruiker("", "", "", "", false, ArrayList<Administratie>())
        }
        val updatedGebruiker = nieuweGebruikerDto.cloneGebruikerWithDtoFields(originalGebruiker)
        crudService!!.updateGebruiker(updatedGebruiker)
        return SingleAnswer("ok")
    }

    @Throws(Exception::class)
    fun removeGebruiker(req: Request, res: Response): Any {
        val gebruiker = SessionHelper.getGebruikerOrThowForbiddenExceptin(req, crudService)
        var gebruikerUuid: String? = req.params(":uuid")
        if ("undefined" == gebruikerUuid) {
            gebruikerUuid = null
        }
        crudService!!.deleteGebruiker(gebruikerUuid)
        return SingleAnswer("ok")
    }

    @Throws(Exception::class)
    fun updatePassword(req: Request, res: Response): Any {
        val gebruiker = SessionHelper.getGebruikerOrThowForbiddenExceptin(req, crudService)
        val gebruikerUuid = req.params(":uuid")
        val newPassword = req.params(":newPassword")
        val gebruikerToChange = crudService!!.getGebruiker(gebruikerUuid)
        val changedGebruiker = gebruikerToChange?.copy(password = newPassword)
        crudService!!.updateGebruiker(changedGebruiker)
        return SingleAnswer("ok")
    }

}
